sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1909,3 +1910,38 @@ def test_fast_sqeuclidean_translation_invariance(
 
     assert_allclose(reference_dist, dist)
     assert_array_equal(reference_nns, nns)
+
+
+@pytest.mark.parametrize("n", [10 ** i for i in [3, 4]])
+@pytest.mark.parametrize("d", [2])
+@pytest.mark.parametrize("ratio_train_test", [10, 1, 0.5])
+@pytest.mark.parametrize("radius", [100, 500])
+@pytest.mark.parametrize("metric", RadiusNeighborhood.valid_metrics())
+def test_fast_radius_neighborhood_reduction_consistency(
+    n,
+    d,
+    ratio_train_test,
+    radius,
+    metric,
+    spread=1000,
+    dtype=np.float64,
+):
+    # Temporary consistency check
+    rng = np.random.RandomState(1)
+
+    X_train = rng.rand(int(n * d)).astype(dtype).reshape((-1, d)) * spread
+    X_test = (
+        rng.rand(int(n * d / ratio_train_test)).astype(dtype).reshape((-1, d)) * spread
+    )
+
+    rn = RadiusNeighborhood.get_for(X=X_test, Y=X_train, radius=radius, metric=metric)
+    dists, indices = rn.compute(return_distance=True, sort_results=True)
+    nn = NearestNeighbors(radius=radius, metric=metric)
+    nn.fit(X_train)
+    reference_dists, references_indices = nn.radius_neighbors(
+        X_test, return_distance=True, sort_results=True
+    )
```

Comment:
At some point in the future,  `NearestNeighbors.radius_neighbors` will rely on this new `RadiusNeighborhood` Cython code and therefore this test will be meaning less.

Would it be possible to instead compare the consistency to naive Python code that uses scipy's `cdist` + basic numpy operations for computing the radius neighbors?

Reply:
Actually, I just saw:

```python
# Temporary consistency check
```

and indeed the existing scikit-learn test suite for `NearestNeighbors.radius_neighbors` is probably enough. So I am fine with keeping this as a temporary test to be removed once `NearestNeighbors` is updated to leverage `RadiusNeighborhood` internally. Maybe add a comment saying so to make that more explicit.

Reply:
Yes, I should have named it "Temporary transitional test".

I will make it more explicit.